strawberryfields/backends/fockbackend/ops.py

```diff
@@ -318,6 +338,109 @@ def apply_gate_einsum(mat, state, pure, modes, n, trunc):
     return np.einsum(einstring, mat, state, mat.conj())
 
 
+def apply_twomode_gate(mat, state, pure, modes, n, trunc, gate="BSgate"):
+    """Applies a two-mode gate to a state
+
+    Applies the two-mode gate to the state using custom tensor contractions and
+    the numba compiler for faster application.
+
+    Args:
+        mat (ndarray): The BS operator to be applied to the state
+        state (ndarray): The state that the BS is applied to
+        pure (bool): If the state is pure or mixed
+        modes (list[int]): A list of modes to which the BS is applied
+        n (int): The total number of modes
+        trunc (int): The Hilbert space truncation/cutoff
+        gate (str): the gate which should be called (BSgate, S2gate)
+
+    Returns:
+        ndarray: State where the two-mode operation has been applied
+    """
+    if pure:
+        t1 = modes[0]
+        t2 = modes[1]
+
+        # put the ket-values in front to be operated on in the apply function
```

Comment:
It might be good to expand on this comment --- what was the previous dimension ordering, and what is the new one? Just to avoid issues with indexing as we had before due to bad documentation!